pkg/scalers/artemis_scaler.go

```diff
@@ -66,10 +66,10 @@ func parseArtemisMetadata(resolvedEnv, metadata, authParams map[string]string) (
 
 	meta.queueLength = defaultArtemisQueueLength
 
-	if val, ok := metadata["restApiTemplate"]; ok && val != "" {
-		meta.restApiTemplate = metadata["restApiTemplate"]
+	if val, ok := metadata["restAPITemplate"]; ok && val != "" {
```

Comment:
this change will break the scaler. `restApiTemplate` is the property that's been expected to be specified in the `ScaledObject` trigger spec

Reply:
https://keda.sh/docs/2.0/scalers/artemis/#trigger-specification

Reply:
Apologies, was excited to open the PR, will update the documentation repo today and open a PR. Thanks for the pointer.

Reply:
No worries :) I am still not sure whether it is good to introduce this particular change to Artemis Scaler. @tomkerkhove wdyt?

Reply:
https://github.com/kedacore/keda-docs/pull/245

Reply:
If it's documented I think it's ok but if we can avoid it I think it's even better.

It's a breaking change which does not provide much added value, but that's just my opinion.

Reply:
I did quick research around disabling golint checks and it's not possible:
https://github.com/golang/lint/issues/263
result of this discussion is freezing the golint:
https://github.com/golang/go/issues/38968

I would say that it may be wise to consider an alternative that is less opinionated (Provides functionality for disabling a specific rule or the entire linter for a file or a range of lines.) and probably has some future:
https://github.com/mgechev/revive

What do you think?

Reply:
The more I am thinking about it, the more I don't like this particular breaking change, as it doesn't bring any benefit for users, just confusion (linting is something that users shouldn't care about, it is our internal stuff).

So the best would be to find a way, how to workaround this particular property.

Reply:
> So the best would be to find a way, how to workaround this particular property.

I will try today to migrate to `revive`. It shouldn't be a big hassle but will allow us to more configuration. 

I didn't know that golint is going to be abandoned :<

Reply:
https://github.com/kedacore/keda/pull/1104

Reply:
@kashifsoofi #1104 is merged so you can rebase this PR on current v2. Then you should be able to run `make revive` to run linting. I suppose that all you changes are good.

In case of this breaking change you may try using comments to disable the check in this one place:
https://github.com/mgechev/revive#comment-directives

Reply:
Just to be clear, `restApiTemplate` change is to be reverted, correct?